django/core/management/commands/dumpdata.py

```diff
@@ -6,6 +8,38 @@
 from django.core.management.utils import parse_apps_and_model_labels
 from django.db import DEFAULT_DB_ALIAS, router
 
+try:
+    import bz2
+    HAS_BZ2 = True
+except ImportError:
+    HAS_BZ2 = False
+
+try:
+    import lzma
+    HAS_LZMA = True
+except ImportError:
+    HAS_LZMA = False
+
+
+def get_open_method(filename, mode='wt'):
+    """Return open method an mode based on the filename suffix."""
+    try:
+        filename_suffix = pathlib.PurePath(filename).suffix
+    except TypeError:
+        filename_suffix = ''
+    compression_formats = {
+        '.gz': (gzip.open, mode),
+    }
+    if HAS_BZ2:
+        compression_formats['.bz2'] = (bz2.open, mode)
+    if HAS_LZMA:
+        compression_formats['.lzma'] = (lzma.open, mode)
```

Comment:
This also requires passing `format=lzma.FORMAT_ALONE` to `lzma.open()` to save in the legacy format.

Reply:
Why should we force to save in the legacy format instead of relying on the default value provided in the lzma stdlib ?

Reply:
Because `.lzma` is the extension for the legacy format. See https://linux.die.net/man/1/lzma

> The native file format is the .xz format, but also the legacy .lzma format and raw compressed streams with no container format headers are supported.

Reply:
I added format for lzma.